### ISSUE TITLE
A question about multiple files of one sample

### ISSUE THREAD

**Issue by NJU-Bio-Info** [MGI-tech-bioinformatics/DNBelab_C_Series_HT_scRNA-analysis-software#87]
Hi, we have sequenced 5 samples' scRNA-seq in your company and when I got the data, I found that there are multiple directories under each sample and for each sub-directory, there are also two pairs of cDNA reads. I'm a little confused about the results, so how can i use dnbc4tools to perform demultiplex and gene expression profile?
![image](https://github.com/user-attachments/assets/843de98a-afc6-439c-b6de-9cb55fc272d2)

or can I just run like this:

dnbc4tools rna run --cDNAfastq1 subdirectory1/cDNA/1_1.fq,subdirectory1/cDNA/2_1.fq,subdirectory2/cDNA/1_1.fq,subdirectory2/cDNA/2_1.fq...?

**Comment by lishuangshuang0616** [comment]
```
dnbc4tools rna run \
--cDNAfastq1 8142-1-211221/cDNA-8142-1-211221/DP8400026224BL_L01_6_1.fq.gz,8142-1-211221/cDNA-8142-1-211221/DP8400026225BL_L01_6_1.fq.gz  \
--cDNAfastq2 8142-1-211221/cDNA-8142-1-211221/DP8400026224BL_L01_6_2.fq.gz,8142-1-211221/cDNA-8142-1-211221/DP8400026225BL_L01_6_2.fq.gz  \
--oligofastq1 8142-1-211221/oligo-8142-1-211221/DP8400026442TR_L01_6_1.fq.gz \
--oligofastq2 8142-1-211221/oligo-8142-1-211221/DP8400026442TR_L01_6_2.fq.gz \
--name 8142-1-211221 \
.......
......
.....
```
like this ,

**Comment by NJU-Bio-Info** [comment]
You mean I can produce 3 results for each sample and then I should integrate them in following downstream analysis like Seurat?

**Comment by lishuangshuang0616** [comment]
What I mean is that we can put the data under each sample together for analysis, and multiple fastq.gz can be put together if they are from the same library. Finally, each sample will get an expression matrix. 
Downstream, Seurat analysis will merge the matrix of multiple samples or remove batch effects.
https://github.com/MGI-tech-bioinformatics/DNBelab_C_Series_HT_scRNA-analysis-software/blob/version2.0/doc/scRNA_para.md
![image](https://github.com/user-attachments/assets/aae1f612-b9a2-46aa-86f7-91226532639e)

**Comment by NJU-Bio-Info** [comment]
Yeah, I understand your .md file so you mean that the 3 sub-directories under each sample are from different experiment? So I can not simply pool them together when I use dnbc4tools and I should analyze them individually and then use maybe Seurat, harmony etc to integrate them together to get the whole results for each sample?

**Comment by lishuangshuang0616** [comment]
You can confirm with the service provider whether the multiple directories under each sample are the results of multiple sequencing runs from the same library of the same sample. 
I'm not sure if you should separate them or merge them. 
Only multiple sequencing data from the same library of the same sample can be merged.

**Comment by NJU-Bio-Info** [comment]
OK, I will check it by myself and thanks for your quick reply! If I know that the 3 sub-directories came from the same sequencing library and then I can merge them, right?

**Comment by lishuangshuang0616** [comment]
yes

**Comment by NJU-Bio-Info** [comment]
maybe just by `zcat` in shell to merge data from different run and same library? Oh, tx you again, the results are very confusing after I downloaded the data.....

**Comment by TroyLi99** [comment]
> maybe just by `zcat` in shell to merge data from different run and same library? Oh, tx you again, the results are very confusing after I downloaded the data.....

hi! did you try use 'zcat' to merge data? did it work?

**Comment by evewu2002** [comment]
> ```
> dnbc4tools rna run \
> --cDNAfastq1 8142-1-211221/cDNA-8142-1-211221/DP8400026224BL_L01_6_1.fq.gz,8142-1-211221/cDNA-8142-1-211221/DP8400026225BL_L01_6_1.fq.gz  \
> --cDNAfastq2 8142-1-211221/cDNA-8142-1-211221/DP8400026224BL_L01_6_2.fq.gz,8142-1-211221/cDNA-8142-1-211221/DP8400026225BL_L01_6_2.fq.gz  \
> --oligofastq1 8142-1-211221/oligo-8142-1-211221/DP8400026442TR_L01_6_1.fq.gz \
> --oligofastq2 8142-1-211221/oligo-8142-1-211221/DP8400026442TR_L01_6_2.fq.gz \
> --name 8142-1-211221 \
> .......
> ......
> .....
> ```
> 
> like this ,

请问同一个样本有多个cdna.fq.gz文件时是否无法使用rna multi命令生成脚本？因为我编辑的sample.tsv完全是按照这个格式的，但是显示IndexError: list index out of range
![image](https://github.com/user-attachments/assets/fbc34766-f638-4b34-a88c-ea6a90269b0f)
